api/app/Policies/UserPolicy.php

```diff
@@ -191,11 +196,12 @@ protected function teamAbleToCheck(User $actor, string $roleId, string $teamId)
                 return $actor->isAbleTo('assign-any-teamRole');
             case 'process_operator':
                 // Community roles have the update-team-processOperatorMembership permission, and it should give them the ability to assign processOperator roles to pools in their community.
-                $communityTeam = $team?->teamable?->team;
+                // for assigning a process, team is a poolTeam so need to reach the community teamable for community checks
+                $poolTeam = $team->loadMissing(['teamable.community.team']);
 
                 return $actor->isAbleTo('update-any-processOperatorMembership')
                     || $actor->isAbleTo('update-team-processOperatorMembership', $team)
-                || (isset($communityTeam) && $actor->isAbleTo('update-team-processOperatorMembership', $communityTeam));
+                || (isset($poolTeam->teamable->community->team) && $actor->isAbleTo('update-team-processOperatorMembership', $poolTeam->teamable->community->team));
```

Comment:
A little unpleasant, but given the `updateUserRoles` mutation passes in a `teamId` that could be pool or community related things are a bit complicated.